scripts/linter/plugins/required-frontmatter.js

```diff
@@ -0,0 +1,45 @@
+const ruleId = "stumptown-linter:frontmatter-required-keys";
+
+/**
+ * If a tree has `tree.data.recipe` (i.e., it's a Markdown doc with a
+ * known-valid recipe), then log messages if it's missing essential frontmatter.
+ */
+function attacher() {
+    return function transformer(tree, file) {
+        if (tree.data && tree.data.recipe) {
+            const { yaml } = tree.children[0].data;
+            const requiredFrontmatter = [];
+
+            // Right now, there's no way to collect the required frontmatter
+            // keys from the recipe itself, so they're hardcoded for now.
```

Comment:
Can't you look at `meta.` keys that don't have a `?` suffix? This will fail because we don't yet understand `info_box`, but it might be better to do this and comment `info_box` out of the recipe. Or is there some other reason this won't work?

Reply:
This would partially work. `examples` would be fixed by checking the recipe. But `attributes` is a weird one: it requires some nested keys that aren't in the recipe. `global` is conventional, though best as I can tell it's not used to build any JSON yet; `element_specific` is used in `build-json` but the only place it's described is in `scripts/build-json/build-recipe-page-json.js`.

There's a little bit of tension here in that I want to check that some frontmatter _exists_, but I'm not ready to build out complete checks for well-formed frontmatter (mostly because I don't think we actually know what well formed is yet).

Reply:
OK, that sounds reasonable. I've changed it to check the recipe for top-level required frontmatter and ignore the nested keys. We can tackle the finer details (e.g., checking for valid spec URLs) in other PRs. Want to take another look?